addon/components/high-charts.js

```diff
@@ -39,40 +39,53 @@ export default Component.extend({
   didReceiveAttrs() {
     this._super(...arguments);
 
-    if (!(get(this, 'content') && get(this, 'chart'))) {
+    const { content, chart, mode } = this.getProperties('content', 'chart', 'mode');
+    if (!content || !chart) {
       return;
     }
 
-    let chart = get(this, 'chart');
     let noData = chart.get('noData');
-
     if (noData != null) {
       noData.remove();
     }
 
-    let numToRemove = chart.series.length - get(this, 'content').length;
-
-    for (let i = numToRemove; i > 0; i--) {
+    const contentSeriesMap = content.reduce((contentSeriesMap, contentSeries) => {
+      contentSeriesMap[contentSeries.name] = contentSeries;
+      return contentSeriesMap;
+    }, {});
 
-      let lastIndex = chart.series.length - 1;
+    // remove and update current series
+    const chartSeriesToRemove = [];
+    chart.series.forEach((series) => {
+      if (series.name === 'Navigator' && mode === 'StockChart') {
+        return;
+      }
 
-      if (chart.series[lastIndex]) {
-        chart.series[lastIndex].remove(false);
+      const contentSeries = contentSeriesMap[series.name];
+      if (!contentSeries) {
+        return chartSeriesToRemove.push(series);
       }
 
-    }
+      series.setData(contentSeries.data, false, false, false);
```

Comment:
What is the reason for disabling animation for `setData()`? I'm curious why you changed it from `series.setData(contentSeries.data, false)`.

API: http://api.highcharts.com/highstock#Series.setData